### PR TITLE
Added basic pause menu functionality

### PR DESCRIPTION
**Pull request by Rokkii**
Pause menu added to test scene 2

### CODE REVIEW

File: UnityClient/Assets/Scripts/PauseMenu.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PauseMenu : MonoBehaviour
+{
+
+    public static bool GamePaused = false; /*Declare GamePaused variable, initially set to false*/
```
Comment (RubyNova) [inline UnityClient/Assets/Scripts/PauseMenu.cs]: `static` is innately evil. Please rework this to not be static. Also public fields are bad practice. Please revise into a full or auto property.

File: UnityClient/Assets/Scripts/PauseMenu.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    public GameObject pauseMenuUI;
+
+    // Update is called once per frame
+    void Update()
+    {
+
+        if (Input.GetKeyDown(KeyCode.Escape))
```
Comment (RubyNova) [inline UnityClient/Assets/Scripts/PauseMenu.cs]: Invert this `if` statement please.

File: UnityClient/Assets/Scripts/PauseMenu.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PauseMenu : MonoBehaviour
+{
+
+    public static bool GamePaused = false; /*Declare GamePaused variable, initially set to false*/
+
+    public GameObject pauseMenuUI;
```
Comment (RubyNova) [inline UnityClient/Assets/Scripts/PauseMenu.cs]: Please rework this public field.

File: UnityClient/Assets/Scripts/PauseMenu.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    // Update is called once per frame
+    void Update()
```
Comment (RubyNova) [inline UnityClient/Assets/Scripts/PauseMenu.cs]: Accessor is missing. Please add it.

**Review by RubyNova — CHANGES_REQUESTED** [review]
Some formatting and structural concerns, also the commands a little bit out-of-place IMO but you can leave those alone if you want.

**Review by Rokkii — COMMENTED** [review]
Code fixes - review request

File: UnityClient/Assets/Scripts/PauseMenu.cs [hunk 5]
```diff
@@ -5,23 +5,23 @@
 public class PauseMenu : MonoBehaviour
 {
 
-    public static bool GamePaused = false; /*Declare GamePaused variable, initially set to false*/
+    private bool GamePaused = false; /*Declare GamePaused variable, initially set to false*/
 
     public GameObject pauseMenuUI;
```
Comment (RubyNova) [inline UnityClient/Assets/Scripts/PauseMenu.cs]: This is not resolved.

File: UnityClient/Assets/Scripts/PauseMenu.cs [hunk 6]
```diff
@@ -5,23 +5,23 @@
...
 
     // Update is called once per frame
-    void Update()
+    private void Update()
     {
 
-        if (Input.GetKeyDown(KeyCode.Escape))
+        if (!Input.GetKeyDown(KeyCode.Escape))
```
Comment (RubyNova) [inline UnityClient/Assets/Scripts/PauseMenu.cs]: This is not resolved.

**Review by RubyNova — CHANGES_REQUESTED** [review]

File: UnityClient/Assets/Scripts/PauseMenu.cs [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+public class PauseMenu : MonoBehaviour
+{
+
+    private bool GamePaused = false; /*Declare GamePaused variable, initially set to false*/
+
+    [SerializeField]
+    private GameObject pauseMenuUI;
+
+    [SerializeField]
+    private void Resume()
```
Comment (RubyNova) [inline UnityClient/Assets/Scripts/PauseMenu.cs:14]: This attribute does nothing for methods. Clue's in the name `SerializeField`. Methods are not fields. Please remove.

File: UnityClient/Assets/Scripts/PauseMenu.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        pauseMenuUI.SetActive(false); /*Disable pause menu UI*/
+        Time.timeScale = 1f; /*Set time in game to stop*/
+        GamePaused = false;
+    }
+
+    [SerializeField]
+    private void Pause()
```
Comment (RubyNova) [inline UnityClient/Assets/Scripts/PauseMenu.cs:22]: Please remove attribute.

**Review by RubyNova — CHANGES_REQUESTED** [review]

File: UnityClient/Assets/Scripts/PauseMenu.cs [hunk 9]
```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PauseMenu : MonoBehaviour
+{
+
+    private bool GamePaused = false; /*Declare GamePaused variable, initially set to false*/
```
Comment (RubyNova) [inline UnityClient/Assets/Scripts/PauseMenu.cs]: This should be name `_likeThis` not `LikeThis`.

File: UnityClient/Assets/Scripts/PauseMenu.cs [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+    [SerializeField]
+    private GameObject pauseMenuUI;
```
Comment (RubyNova) [inline UnityClient/Assets/Scripts/PauseMenu.cs]: Missing underscore prefix for a private field. Please fix.

**Review by RubyNova — CHANGES_REQUESTED** [review]
very minor adjustments then good to merge.

**Review by RubyNova — APPROVED** [review]
LGTM.

**Pull request merged**